static/app/views/alerts/rules/teamFilter.tsx

```diff
@@ -57,14 +69,17 @@ function TeamFilter({
       filtered: false,
     },
   ];
-  const teamItems = teams.map(({id, slug}) => ({
-    label: slug,
-    value: id,
-    filtered: teamFilterSearch
-      ? !slug.toLowerCase().includes(teamFilterSearch.toLowerCase())
-      : false,
-    checked: selectedTeams.has(id),
-  }));
+  const isMemberTeams = teams.filter(team => team.isMember);
+  const teamItems = (isSuperuser ? teams : showIsMemberTeams ? isMemberTeams : teams).map(
+    ({id, slug}) => ({
+      label: slug,
+      value: id,
+      filtered: teamFilterSearch
+        ? !slug.toLowerCase().includes(teamFilterSearch.toLowerCase())
+        : false,
+      checked: selectedTeams.has(id),
+    })
+  );
```

Comment:
Do we want to update this component to only display teams the user is a member of? This currently shows all teams (even if not a superuser), which is confusing.